### PR TITLE
better check of crdt/datatype by looking at props & bucket, as per wo… [JIRA: RIAK-2532]

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
…rk in feature-zl-add_expanded_is_crdt
- [x] Relies on review/merge of https://github.com/basho/riak_kv/pull/1405 - RIAK-2531 - for the Green!

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
create jira issue (a conundrum)

File: src/yz_kv.erl [hunk 1]
```diff
@@ -578,30 +580,49 @@ siblings_permitted_test_() ->
...
                  ?assertEqual(counter, proplists:get_value(datatype, BTProps1)),
                  ?assertEqual(map, proplists:get_value(datatype, BTProps2)),
+                 ?assertEqual(undefined, proplists:get_value(datatype,
+                                                             BTProps3)),
                  [begin
                       Object = riak_object:new(B, K, V),
                       CheckBucket = riak_object:bucket(Object),
                       CheckBucketProps = riak_core_bucket:get_bucket(CheckBucket),
-                      ?assertNot(siblings_permitted(CheckBucketProps))
+                      ?assert(siblings_permitted(Object, CheckBucketProps))
```
Comment (fadushin) [inline src/yz_kv.erl:612]: I don't quite understand why the logic is reversed in this check.  The objects in question are in a counter and map bucket, respectively.  Does the is_datatype check (i.e., riak_kv_crdt:is_crdt/2) return a different value in the most recent riak_kv version?

Comment (zeeshanlakhani) [inline src/yz_kv.erl:612]: discussed over SH.

Comment (fadushin) [inline src/yz_kv.erl:612]: Thanks @zeeshanlakhani.  As I understand it, there are entries that can be in a CRDT bucket that are not CRDTs, proper, and can therefore have siblings.  The siblings_permitted (or more specifically, the riak_kv_crdt:is_crdt/2 has been refined to catch this condition, hence the logic change in the code and therefore in the test.

**Comment by fadushin** [comment]
+1 5d06ece

**Comment by fadushin** [comment]
@borshop merge

**Comment by Basho-JIRA** [comment]
Already merged, close per Doug 

**_[posted via JIRA by Patricia Brewer]**_

**Pull request merged**